### PR TITLE
rename control plane & tenant cluster

### PR DESCRIPTION
**Pull request by TheoBrigitte**
Towards: https://github.com/giantswarm/giantswarm/issues/15028

- renamed some alerts
- move controller/control-plane > controller/managementcluster
- removed control plane or tenant cluster notation in some places

### CODE REVIEW

File: helm/prometheus-meta-operator/templates/recording-rules/service-levels.rules.yml [hunk 1]
```diff
@@ -3,7 +3,7 @@ kind: PrometheusRule
 metadata:
   labels:
     {{- include "labels.common" . | nindent 4 }}
-    cluster_type: "control_plane"
+    cluster_type: "management_clustemanagement_cluster"
```
Comment (QuentinBisson) [inline helm/prometheus-meta-operator/templates/recording-rules/service-levels.rules.yml:6]:
```suggestion
    cluster_type: "management_cluster"
```

File: service/controller/managementcluster/controller.go [hunk 2]
```diff
@@ -73,7 +73,7 @@ func NewController(config ControllerConfig) (*Controller, error) {
 		c := controller.Config{
 			K8sClient: config.K8sClient,
 			Logger:    config.Logger,
-			Name:      project.Name() + "-control-plane-controller",
+			Name:      project.Name() + "-management-cluster-controller",
```
Comment (QuentinBisson) [inline service/controller/managementcluster/controller.go:76]: We will need to be careful here as we will need to remove the finalizers

**Comment by QuentinBisson** [comment]
Did you update the tests as well?

File: helm/prometheus-meta-operator/templates/recording-rules/service-levels.rules.yml (hunk 1 above)
Comment (TheoBrigitte) [reply helm/prometheus-meta-operator/templates/recording-rules/service-levels.rules.yml:6]: nice typo, thanks.

File: service/controller/managementcluster/controller.go (hunk 2 above)
Comment (TheoBrigitte) [reply service/controller/managementcluster/controller.go:76]: oh gosh, almost forgot about those. I guess we can keep a tiny `control-plane-controller` which only delete finalizers.

Comment (QuentinBisson) [reply service/controller/managementcluster/controller.go:76]: Is there a way in operatorkit to have a controller that only removes finalizers without adding them?

Comment (TheoBrigitte) [reply service/controller/managementcluster/controller.go:76]: I don't know

**Comment by TheoBrigitte** [comment]
> Did you update the tests as well?

done

**Review by QuentinBisson — APPROVED** [review]

File: service/controller/managementcluster/controller.go (hunk 2 above)
Comment (TheoBrigitte) [reply service/controller/managementcluster/controller.go:76]: added legacy finalizer resource, which takes care of removing the old finalizers.

**Pull request merged**